src/gravel/tests/conftest.py

```diff
@@ -85,7 +91,9 @@ def ceph_conf_file_fs(
 
 
 @pytest.fixture()
-def get_data_contents(fs: fake_filesystem.FakeFilesystem):
+def get_data_contents(  # type: ignore
+    fs: fake_filesystem.FakeFilesystem
+) -> str:
```

Comment:
```suggestion
def get_data_contents(
    fs: fake_filesystem.FakeFilesystem
) -> Generator[Callable[[str, str], str], None, None]:
```

these ones are fun, but I think the above is correct

Reply:
I think you're correct, but eww :-p

Reply:
tried that. It works to shut up mypy, but won't shut up pyright. That's because when it's called, it's not called as a generator, butt as a function (because pytest is doing magic in the background). So pyright will be annoyed because we're calling on a generator without actually using it as a generator.

TL;DR: pytest typing support sucks, and this was the closest I could get to make that function work.